src/sql/SqlService.ts

```diff
@@ -0,0 +1,405 @@
+            timeoutMillis = Long.fromNumber(sqlStatement.options.timeoutMillis);
+        } else {
+            timeoutMillis = SqlServiceImpl.DEFAULT_TIMEOUT;
+        }
+
+        try {
+            const serializedParams = new Array(sqlStatement.params.length);
+            if (Array.isArray(sqlStatement.params)) { // params can be undefined
+                for (let i = 0; i < sqlStatement.params.length; i++) {
+                    serializedParams[i] = this.serializationService.toData(sqlStatement.params[i]);
+                }
+            }
+            const cursorBufferSize = sqlStatement.options?.hasOwnProperty('cursorBufferSize') ?
+                sqlStatement.options.cursorBufferSize : SqlServiceImpl.DEFAULT_CURSOR_BUFFER_SIZE;
+
+            const returnRawResult = sqlStatement.options?.hasOwnProperty('returnRawResult') ?
+                sqlStatement.options.returnRawResult : SqlServiceImpl.DEFAULT_FOR_RETURN_RAW_RESULT;
+
+            const schema = sqlStatement.options?.hasOwnProperty('schema') ?
+                sqlStatement.options.schema : SqlServiceImpl.DEFAULT_SCHEMA;
+
+            const requestMessage = SqlExecuteCodec.encodeRequest(
+                sqlStatement.sql,
+                serializedParams,
+                timeoutMillis,
+                cursorBufferSize,
+                schema,
+                expectedResultType,
+                queryId
+            );
+
+            const res = SqlResultImpl.newResult(
+                this,
+                this.serializationService,
+                connection,
+                queryId,
+                cursorBufferSize,
+                returnRawResult,
+                this.connectionManager.getClientUuid()
+            );
+
+            this.invocationService.invokeOnConnection(connection, requestMessage).then(clientMessage => {
+                SqlServiceImpl.handleExecuteResponse(clientMessage, res);
+            }).catch(err => {
+                res.onExecuteError(
+                    this.toHazelcastSqlException(err, connection)
+                );
+            });
+
+            return res;
+        } catch (error) {
+            throw this.toHazelcastSqlException(error, connection);
+        }
+    }
+
+    execute(sql: string, params?: any[], options?: SqlStatementOptions): SqlResult {
+        let sqlStatement: SqlStatement;
+
+        if (typeof sql === 'string') {
+            sqlStatement = {
+                sql: sql
+            };
+            // If params is not provided it won't be validated. Default value for optional parameters is undefined.
+            // So, if they are undefined we don't set it, and in validator method we check for property existence.
+            if (params !== undefined) {
+                sqlStatement.params = params;
+            }
+            if (options !== undefined) {
+                sqlStatement.options = options;
+            }
+        } else if (typeof sql === 'object') {
```

Comment:
Can it be an object anymore? If this is the case, the usage is wrong and we shouldn't permit it